hyperloglogplus.go

```diff
@@ -235,7 +235,11 @@ func (h *HyperLogLogPlus) estimateBias(est float64) float64 {
 func (h *HyperLogLogPlus) Count() uint64 {
 	if h.sparse {
 		h.mergeSparse()
-		return uint64(linearCounting(mPrime, mPrime-uint32(h.sparseList.Count)))
+		if uint32(h.sparseList.Len()) > h.m {
```

Comment:
i think it might make sense to put the `uint32(h.sparseList.Len()) > h.m` check at the end of mergeSparse, and then make sure to convert to normal there. Then if mergeSparse is ever used somewhere else, this should be safe.